### ISSUE TITLE
Backend refactoring

### ISSUE THREAD

**Issue by confluence** [CARTAvis/carta-backend#959]
This epic collects all outstanding issues related to backend code refactoring and code style.

**Comment by pford** [comment]
I just ran across the backend Timer class which was modified slightly in a PR but does not seem to be used in the code.  Should this be added to the list (use-it-or-lose-it)?  It would make the performance logging code throughout the backend look nicer but adds overhead for map access.

And speaking of maps, should we prefer map::find over map::count to check for a given key and make this consistent overall?

**Comment by confluence** [comment]
Agreed on both counts -- we should make a list of classes and functions which are no longer used. And I think `find` is more idiomatic for checking for a key.

**Comment by pford** [comment]
I stand corrected, the carta::Timer will be used by the new C++ ICD tests when performance is enabled.
